mysys/my_gethwaddr.c

```diff
@@ -87,11 +87,15 @@ my_bool my_gethwaddr(uchar *to)
 my_bool my_gethwaddr(uchar *to)
 {
   int fd, res= 1;
+#ifdef _AIX
+  struct ifhwaddr_req ifr[32];
+#else
   struct ifreq ifr[32];
+#endif
   struct ifconf ifc;
   DBUG_ENTER("my_gethwaddr");
 
-  ifc.ifc_req= ifr;
+  ifc.ifc_req= (struct ifreq *) ifr;
```

Comment:
This looks a bit suspicious . Is it safe to cast ifhwaddr_req to struct ifreq ?
I did a web search for ifhwaddr_req, but could not find anything.
Can you post the declaration of ifhwaddr_req so that I can check that it is compatible with struct ifreq?

Reply:
```
/* ifreq structure for SIOCGIFHWADDR */
struct ifhwaddr_req {
        char    ifr_name[IFNAMSIZ];               /* if name, e.g. "en0" or "ib0" */
        char    ifr_hwaddr[MAX_HWADDR_LEN];          /* device hardware address */
        int     addr_len;
};
```

Reply:
```
struct  ifreq {
#ifndef IFNAMSIZ        /* Also in net_if.h */
#define IFNAMSIZ        16
#endif
        char    ifr_name[IFNAMSIZ];             /* if name, e.g. "en0" */
        union {
                struct  sockaddr ifru_addr;
                struct  sockaddr ifru_dstaddr;
                struct  sockaddr ifru_broadaddr;
                __ulong32_t     ifru_flags;
                int     ifru_metric;
                caddr_t ifru_data;
                u_short ifru_site6;
                __ulong32_t   ifru_mtu;
                int     ifru_baudrate;
                int     ifru_checksum_offload[2];
        } ifr_ifru;
...
}
```

Reply:
it is probably ok.  Hard to say, but if the code was copied from some old implementation then we can probably rely on it.
Just to be safe, where did you get the information that one should use struct ifhwaddr_req instead of struct ifreq on AIX?